eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php

```diff
@@ -628,46 +639,122 @@ public function locationSwapped($location1Id, $location1ParentId, $location2Id,
         $autoLocation1 = $this->gateway->loadAutogeneratedEntry("eznode:{$location1Id}");
         $autoLocation2 = $this->gateway->loadAutogeneratedEntry("eznode:{$location2Id}");
 
+        $contentInfo1 = $this->contentGateway->loadContentInfoByLocationId($location1Id);
+        $contentInfo2 = $this->contentGateway->loadContentInfoByLocationId($location2Id);
+
+        $names1 = $this->getNamesForAllLanguages($contentInfo1);
+        $names2 = $this->getNamesForAllLanguages($contentInfo2);
+
+        $alwaysAvailable1 = (bool)($contentInfo1['language_mask'] & 1);
```

Comment:
@alongosz Is there an better way to check this "always available" flag? I mean constant representing this flag or method  which might be used to check this.

Reply:
>  Is there an better way to check this "always available" flag? I mean constant representing this flag or method which might be used to check this.

Hmm, here it was hardcoded from the beginning. I think that maybe we should start relying more on `\eZ\Publish\Core\Persistence\Legacy\Content\Language\MaskGenerator::isAlwaysAvailable`.